packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -609,6 +614,36 @@ export class EntityService {
     return true;
   }
 
+  async updateFieldDataTypeIdByRelatedEntity(
+    field: EntityField,
+    relatedEntityId: string
+  ): Promise<EntityField> {
+    const relatedIdField = await this.prisma.entityField.findFirst({
+      where: {
+        dataType: EnumDataType.Id,
+        entityVersion: {
+          entityId: relatedEntityId,
+        },
+      },
+    });
+
+    const idProp = relatedIdField.properties as unknown as {
+      [key: string]: JsonValue;
+    };
+
+    const idType = idProp["idType"];
+    if (idType === "CUID" || idType === "UUID") {
```

Comment:
just to upgrade:
instead of if, we can do:
const idTypeMap = {
  CUID: EnumDataType.SingleLineText,
  UUID:  EnumDataType.SingleLineText,
  AUTO_INCREMENT: EnumDataType.WholeNumber,
  AUTO_INCREMENT_BIG_INT: EnumDataType.WholeNumber
}
then get: field.dataType = idTypeMap[idType]